doc/task-hierarchy.md

```diff
@@ -0,0 +1,164 @@
+- Title Generation
+- Weblink Generation
+- Text Modification
+- Classification
+- Verification 
+   - Verify if two inputs match with each other e.g. a task to verify if a sentence and its sentiment given in the input match
+- Translation
+- Hypernym Discovery
+- Parts-of-speech
+- Coreference: event coreference
+- Intent Detection
+- Claim Verification
+- Paraphrasing
+- Relation Prediction
+- Analogical Reasoning
+- Semantic Parsing
+- Stance Detection
+- Word Sense Disambiguation
+- Mathematics: Arithmetic
+- Mathematics: Algebra
+- Mathematics: Geometry
+- Question Decomposition
+- Counting
+   - Count an attribute of input e.g. a task to count number of vowels in a given word
+- Entity Detection
+- Role Labelling
+- Combinatorics
+- Grammar Error Correction
+- Summarization
+- Sentiment Analysis
+- Hate Speech Detection
+- Fake News Detection
+- Ethical Judgement
+- Style Transfer
+- Text Comparison
+- Question Composition
+   - Compose questions by concating questions in the input
+- Column Matching
+   - Given two sets in the input, generate a mapping between them e.g. given a set of countries and their capitals in the input, generate an output that maps countries to capitals.
+- Handling Commands
+- Author Identification
+- Sorting
+- Language Identification
+- Text Span Selection
+- Hallucination
+   - Given a context, generate imaginary content e.g. given a sentence, generate a story/poem.
+- Document Classification
+- Relevancy Estimation
+- Named Entity Recognition
+- Dialogue Understanding
+- Abuse Detection
+- Text Simplification
+- Tabular Text Operation
+- Coreference: Entity Coreference
+- Coreference: Pronoun Disambiguation
+- Reasoning: Logic
+- Reasoning: Numerical Reasoning
+- Reasoning: Reasoning on Actions
+- Reasoning: General Commonsense Reasoning
+- Reasoning: Abductive Reasoning
+- Reasoning: Temporal Reasoning
+- Reasoning: Spatial Reasoning
+- Reasoning: Multihop Reasoning
+- Reasoning: Reasoning on Social Interactions
+- Reasoning: Commonsense Reasoning
+   - Tasks related to activities humans do in daily life e.g. eating breakfast in the morning, sleeping during night etc.
+- Reasoning: Numerical Commonsense Reasoning
+   - Tasks which requires numerical commonsense knowledge e.g. a car has 4 wheels.
+- Reasoning: Physical Reasoning
+   - Tasks involving physical interactions with objects e.g. a knife (and not a paper) is used to cut objects
+- Reasoning: Logical Reasoning
+- Reasoning: Planning
+   - Tasks which need some sort of planning e.g. how to go to Hawaii?
+- Reasoning: Qualitative Reasoning
+- Reasoning: Deductive Reasoning
+- Reasoning: Causal Reasoning
+- Reasoning: Reasoning with Symbols
+   - Tasks where symbols represent various things e.g. if X is the number of apples in the freeze today morning and Y is the number remaining after I ate a few apples, X-Y is the number of apples I ate.
+
+Domain:
```

Comment:
Should "dialogue" also be a domain?

Reply:
I think dialogue is too abstract to be a domain. Dialogue can be on various domains e.g. medical, movies, sports etc.

Also, we have a task type: "Dialogue Understanding".

Reply:
I would still suggest having a "Dialogue" as a text domain. The language of say, Alexa conversations are much different than anything else that we have on the list.

Reply:
Added